sdl-core/src/main/scala/io/smartdatalake/metrics/SparkStageMetricsListener.scala

```diff
@@ -19,30 +19,38 @@
 
 package io.smartdatalake.metrics
 
-import io.smartdatalake.config.SdlConfigObject
 import io.smartdatalake.config.SdlConfigObject.{ActionId, DataObjectId}
 import io.smartdatalake.util.misc.SmartDataLakeLogger
-import io.smartdatalake.workflow.action.{Action, SDLExecutionId}
-import io.smartdatalake.workflow.{ActionMetrics, ActionPipelineContext}
-import org.apache.spark.scheduler.{SparkListener, SparkListenerJobStart, SparkListenerStageCompleted}
+import io.smartdatalake.workflow.ActionPipelineContext
+import io.smartdatalake.workflow.action.ActionSubFeedsImpl.MetricsMap
+import io.smartdatalake.workflow.action.SDLExecutionId
+import org.apache.spark.scheduler.{SparkListener, SparkListenerJobEnd, SparkListenerJobStart, SparkListenerStageCompleted}
 
 import scala.collection.mutable
 import scala.util.matching.Regex
 
 /**
- * Collects spark metrics for spark stages.
+ * Collects spark metrics for spark stages filtered by ActionId and an optional DataObjectId
  */
-private[smartdatalake] class SparkStageMetricsListener(action: Action)(implicit context: ActionPipelineContext) extends SparkListener with SmartDataLakeLogger {
+private[smartdatalake] class SparkStageMetricsListener(actionId: ActionId, dataObjectId: Option[DataObjectId] = None)(implicit context: ActionPipelineContext) extends SparkListener with SmartDataLakeLogger {
+
+  /**
+   * Keeps track of running jobs, e.g. jobs that started but did not yet end.
+   * Map key is jobId.
+   */
+  val runningJobs: mutable.Map[Int, JobInfo] = mutable.Map.empty
 
   /**
    * Stores jobID and jobDescription indexed by stage ids.
    */
   val jobInfoLookupTable: mutable.Map[Int, JobInfo] = mutable.Map.empty
```

Comment:
```suggestion
  val stageIdToJobInfo: mutable.Map[Int, JobInfo] = mutable.Map.empty
```
A name like that makes it clear that it's just a map. I found the word "Table" confusing, I needed to open IntelliJ to understand what it does